src/core/detail/bitpacking.cc

```diff
@@ -104,6 +108,33 @@ static inline pair<const char*, uint8_t*> simd_variant2_pack(const char* ascii,
 // See https://github.com/lemire/fastvalidate-utf-8/
 // The function returns true (1) if all chars passed in src are
 // 7-bit values (0x00..0x7F). Otherwise, it returns false (0).
+#ifdef __s390x__
+bool validate_ascii_fast(const char* src, size_t len) {
+  size_t i = 0;
+  vector unsigned char has_error = vec_splat_s8(0);
```

Comment:
hey @iko1 . can you please add comments so that reader unfamiliar with s390 vector code 
will understand the intent.

Reply:
Done.